### PR TITLE
Vulkan backend (Cherry Pick some changes for RC)

### LINKED ISSUE #4274: Vulkan Control?
I see in the upcoming `0.10` release an OpenGL control is included.  To future proof Avalonia better, I think it would be wise to include a sort of Vulkan Control as well.  There are projects out there such as [Silk.NET](https://github.com/Ultz/Silk.NET) which provide cross platform Vulkan bindings (have yet to try them out for myself).

Other widget toolkits such as Qt have [already added Vulkan capability](https://doc.qt.io/qt-5/qvulkanwindow.html).  And I'm sure that more are to follow.

### PR DESCRIPTION
**Pull request by kekekeks**
Mostly based on https://github.com/AvaloniaUI/Avalonia/pull/7753 but uses the new infrastructure introduced in https://github.com/AvaloniaUI/Avalonia/pull/9639

Fixes https://github.com/AvaloniaUI/Avalonia/issues/4274

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027738-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by hez2010** [comment]
If I recalled correctly, Vulkan doesn't support `WS_EX_NOREDIRECTIONBITMAP`. Will this introduce regressions on Windows?

**Comment by kekekeks** [comment]
1) Vulkan backend is optional, apps will still use ANGLE on Windows by default
2) at some point we'll implement WinUI composition output for Vulkan too (we just need to render into a DXGI surface)

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027763-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027775-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027777-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Vulkan/VulkanControlBase.cs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    }
+
+    async Task<bool> InitializeAsync()
+    {
+        var feature = (IVulkanSharedDeviceGraphicsContextFeature?)
+            await this.GetVisualRoot()!.Renderer.TryGetRenderInterfaceFeature(
+                typeof(IVulkanSharedDeviceGraphicsContextFeature));
+        if (feature?.SharedDevice == null)
+        {
+            Logger.TryGet(LogEventLevel.Error, "OpenGL")?.Log("VulkanControlBase",
```
Comment (llfab) [inline src/Avalonia.Vulkan/VulkanControlBase.cs:50]: "OpenGL" => "Vulkan"

File: src/Avalonia.X11/X11Platform.cs [hunk 2]
```diff
@@ -289,6 +298,11 @@ public class X11PlatformOptions
         /// </remarks>
         public bool? EnableMultiTouch { get; set; } = true;
 
+        /// <summary>
+        /// Enables Vulkan rendering backend
+        /// </summary>
+        public bool UseVulkan { get; set; }
```
Comment (llfab) [inline src/Avalonia.X11/X11Platform.cs:304]: This seems to  fit into what is already there. UseXy is present in all kinds of backends. I wonder whether at some point we do not distribute UseXy into all separate XyPlatformOptions. Anyways fits to how it is done already.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027790-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by kekekeks** [comment]
Superceded by #11365 12737

**Pull request closed without merging**